shadow.c

```diff
@@ -198,6 +207,22 @@ PHP_MINIT_FUNCTION(shadow)
 {
 	zend_function *orig;
 
+	// verify that shadow is loaded after other extensions
+	HashPosition pos;
+	zend_module_entry *data = NULL;
+
+	for (zend_hash_internal_pointer_reset_ex(&module_registry, &pos);
+		zend_hash_get_current_data_ex(&module_registry, (void *) &data, &pos) == SUCCESS;
+		zend_hash_move_forward_ex(&module_registry, &pos)
+	) {
+		if (!data->module_started
+			&& strcmp(data->name, "apc") && strcmp(data->name, "mhash") // exceptions
+		) {
+			php_printf("Shadow extension should be the last one loaded. Extension %s is not loaded yet.\n", (char*)data->name);
+			return FAILURE;
```

Comment:
I wouldn't say it's a failure. If it's possible, it should be just a warning. Given that the extension's API is not supposed to be overridden, it's fine if goes after shadow. BTW, it'd be even more useful, if we could catch the function/method which extension registers or overrides, but it's configured to be overridden by shadow.